`Alert`: Fix link colors (HDS-2396) (#1576)
hashicorp/design-system: website/docs/components/alert/partials/content/content.md

> ## General recommendations

Reviewer: @jorytindall similar for the code showcase, this content (or part of it) should also be replicated/adapted for the `Toast` component, that inherits all the behaviours/functionalities/styles from the `Alert`.

Reviewer 2: Ok, I've made several updates to the content, broken down the recommendations a bit. Let me know what you think, if it works I can translate/copy relevant content to the Toast.

Reviewer: It works™

- Keep the title short, as this will be the most prominent element when users scan the alert.
- Avoid ending the title with a period.
- Alert descriptions should be short but clear enough to explain what’s happening. We recommend keeping messages under 90 characters.
- For warning and critical alerts, guide the users on how to prevent or fix the issue.

## Actions

Use small buttons to avoid competing with other actions on the page. Use more than two actions sparingly.

We recommend using the `secondary` button variant for primary actions and the `tertiary` button variant for secondary actions.

<Hds::Alert @type="inline" as |A|>
  <A.Title>Recommended button usage</A.Title>
  <A.Description>Lorem ipsum dolar sit amet.</A.Description>
  <A.Button @text="Your action" @color="secondary" @onClick={{this.noop}} />
  <A.Button @color="tertiary" @icon="arrow-right" @iconPosition="trailing" @text="Tertiary" />
</Hds::Alert>

For content guidelines on actions, refer to [Button](/components/button) and 
[Link](/components/link/standalone) documentation.

### Usage of critical buttons

> Avoid using critical buttons in alerts. We handle the prominence and importance via the styling of the alert container itself. If needing to confirm that the user intended to interact with the action, consider displaying a confirmation modal.

Author: ```suggestion
Avoid using critical buttons in alerts. We handle the prominence and importance via the styling of the alert container itself. If you need to confirm that the user intended to interact with the action, consider displaying a confirmation modal.
```

!!! Dont
<Hds::Alert @type="inline" @color="critical" as |A|>
  <A.Title>Critical button usage</A.Title>
  <A.Description>Lorem ipsum dolar sit amet.</A.Description>
  <A.Button @text="Your action" @color="critical" @onClick={{this.noop}} />
</Hds::Alert>
!!!

> ## Links

Reviewer 3: Within the Alert components, "Actions" refer to both Buttons and Links within the "Footer" (although not defined as a footer) of the component. For this reason, I recommend nesting the "Links" section under the "Actions" section and keeping it related to the standalone link found in the "Footer." You can then move all new content about links within the description to a new section (H2) "Description." This may also be more scalable because it allows us to add more content about the description later. 

So essentially, the new content structure would look something like this: 

- Actions
  - Buttons
    - Usage of critical buttons
  - Links
    - Within page and inline Alerts
- Description

Reviewer 2: Got it, makes sense

> ### Links

Reviewer: While reading this part, I was a bit confused on what links you (@jorytindall?) were referring to. IN some cases we're referring to the "actions/buttons" (that technically can be links as in `HTML anchors`) and in other cases to the inline links within the description text.
This I suspect may lead to confusion. For example, a user may think that it's OK to use a "standalone" link in the description (which is not).
Not sure how to improve this, but I think is worth another cycle of refinement.

Reviewer 2: Yeah, that's a good identification of the challenge. I'll rev on this again

Reviewer 2: I added a bit of clarity on a different comment, but it's applicable. Let me try to revise everything holistically.

Reviewer: Perfect! 🙌

### Within page and inline Alerts

Use [Standalone Link](/components/link/standalone) when an action takes the user to a new destination (URL).

We recommend using the secondary standalone link to maintain visual hierarchy and avoid competing prominence when used with the secondary button.

<Hds::Alert @type="inline" @color="highlight" as |A|>
  <A.Title>Links in alerts</A.Title>
  <A.Description>Lorem ipsum dolar sit amet.</A.Description>
  <A.Button @text="Your action" @color="secondary" @onClick={{this.noop}} />
  <A.Link::Standalone @icon="arrow-right" @iconPosition="trailing" @color="secondary" @text="Standalone link" @href="#" />
</Hds::Alert>

> If multiple links are necessary or if linking within the Alert description, use the `secondary` [Inline Link](/components/link/inline).

Reviewer 2: On second thought, this seems a bit irrelevant/confusing now. Should be removed?

> If multiple links are necessary or if linking within the Alert description, use the `secondary` [Inline Link](/components/link/inline).

Reviewer 3: I think the part about links in the description is the update that @KristinLBradley was looking for us to add, so that still feels valuable.

> If multiple links are necessary or if linking within the Alert description, use the `secondary` [Inline Link](/components/link/inline).

Reviewer 2: Simplified this language a bit to focus on the description portion of the Alert.

<Hds::Alert @type="inline" @color="warning" as |A|>
  <A.Title>Multiple links</A.Title>
  <A.Description>
    Lorem ipsum dolor sit amet, <Hds::Link::Inline @href="#" @color="secondary">consectetur</Hds::Link::Inline> adipiscing elit, sed do <Hds::Link::Inline @href="#" @color="secondary">eiusmod</Hds::Link::Inline> tempor incididunt ut <Hds::Link::Inline @href="#" @color="secondary">labore</Hds::Link::Inline> et dolore magna aliqua.
  </A.Description>
</Hds::Alert>

> If linking to multiple related items on the same page, e.g., highlighting multiple form validation errors, wrap the links in an list.

Author: ```suggestion
If linking to multiple related items on the same page, e.g., highlighting multiple form validation errors, wrap the links in a list.
```

<Hds::Alert @type="inline" @color="critical" as |A|>
  <A.Title>Error submitting the form</A.Title>
  <A.Description>
    The follow fields must be corrected before submitting the form:
    <li class="hds-typography-body-200">
      <Hds::Link::Inline @color="secondary" @href="#">Email address</Hds::Link::Inline>
    </li>
    <li class="hds-typography-body-200">
      <Hds::Link::Inline @color="secondary" @href="#">First name</Hds::Link::Inline>
    </li>
    <li class="hds-typography-body-200">
      <Hds::Link::Inline @color="secondary" @href="#">Billing address</Hds::Link::Inline>
    </li>
  </A.Description>
</Hds::Alert>

### Within compact Alerts

We recommend using `secondary` [Inline Links](/components/link/inline) in the `compact` Alert to match the intended hierarchy of the component and when linking to multiple URLs.

<Hds::Alert @type="compact" @color="success" as |A|>
  <A.Description>

>     You've successfully configured your account. Next, <Hds::Link::Inline @href="#" @color="secondary">create a cluster</Hds::Link::Inline>, <Hds::Link::Inline @href="#" @color="secondary">invite your teammates</Hds::Link::Inline>, or create a <Hds::Link::Inline @href="#" @color="secondary">HashiCorp Virtual Network (HVN)</Hds::Link::Inline>.

Author: ```suggestion
    You’ve successfully configured your account. Next, <Hds::Link::Inline @href="#" @color="secondary">create a cluster</Hds::Link::Inline>, <Hds::Link::Inline @href="#" @color="secondary">invite your teammates</Hds::Link::Inline>, or create a <Hds::Link::Inline @href="#" @color="secondary">HashiCorp Virtual Network (HVN)</Hds::Link::Inline>.
```

  </A.Description>
</Hds::Alert>

> If the link requires more prominence, consider using the `primary` color.

Reviewer: It's not clear if this applies also to the `Page` and `Inline` variants of the `Alert`, and –if not– if it's simply not recommended or "forbidden".

Reviewer 2: This is directly under the headline "Within compact alerts", so I would expect it only apply to compact alerts. But maybe it can made clearer by outlining something like this:

- In a `Compact` alert: we recommend using secondary inline links within the description.
    - It's use a primary link if it requires more prominence, but don't do this if there are multiple links within the description.
 - In `Page` and `Inline` alerts: if linking within the description of the component, use secondary inline links.
    -  If the link requires more prominence, use a standalone link in the actions/footer.

Reviewer: Works well now. 👍

<Hds::Alert @type="compact" @color="success" as |A|>
  <A.Description>

>     You've successfully configured your account. Next, <Hds::Link::Inline @href="#" @color="primary">create your first cluster</Hds::Link::Inline>.

Author: ```suggestion
    You’ve successfully configured your account. Next, <Hds::Link::Inline @href="#" @color="primary">create your first cluster</Hds::Link::Inline>.
```

  </A.Description>
</Hds::Alert>

!!! Dont

Don’t mix and match different link colors in the same Alert.

<Hds::Alert @type="compact" @color="success" as |A|>
  <A.Description>

>     You've successfully configured your account. Next, <Hds::Link::Inline @href="#" @color="primary">create a cluster</Hds::Link::Inline>, <Hds::Link::Inline @href="#" @color="secondary">invite your teammates</Hds::Link::Inline>, or create a <Hds::Link::Inline @href="#" @color="secondary">HashiCorp Virtual Network (HVN)</Hds::Link::Inline>.

Author: ```suggestion
    You’ve successfully configured your account. Next, <Hds::Link::Inline @href="#" @color="primary">create a cluster</Hds::Link::Inline>, <Hds::Link::Inline @href="#" @color="secondary">invite your teammates</Hds::Link::Inline>, or create a <Hds::Link::Inline @href="#" @color="secondary">HashiCorp Virtual Network (HVN)</Hds::Link::Inline>.
```

[Link](/components/link/standalone) documentation.

### Usage of critical buttons

Avoid using critical buttons in alerts. We handle the prominence and importance via the styling of the alert container itself. If you need to confirm that the user intended to interact with the action, consider displaying a confirmation modal.

!!! Dont
<Hds::Alert @type="inline" @color="critical" as |A|>
  <A.Title>Critical button usage</A.Title>
  <A.Description>Lorem ipsum dolar sit amet.</A.Description>
  <A.Button @text="Your action" @color="critical" @onClick={{this.noop}} />
</Hds::Alert>
!!!

### Links

#### Within page and inline Alerts

When linking to internal and external resources in the Alert, default to using a `secondary` [Standalone Link](/components/link/standalone) in the `actions` area. In the Ember component, these elements are passed as [contextual components](?tab=code#actions).

<Hds::Alert @type="inline" @color="highlight" as |A|>
  <A.Title>Links in alerts</A.Title>
  <A.Description>Lorem ipsum dolar sit amet.</A.Description>
  <A.Link::Standalone @icon="arrow-right" @iconPosition="trailing" @color="secondary" @text="Standalone link" @href="#" />
</Hds::Alert>

> If linking within the Alert description, use the `secondary` [Inline Link](/components/link/inline) as the default. A common use case for this is when linking to multiple resources.

Author: ```suggestion
Within the Alert description, use the `secondary` [Inline Link](/components/link/inline) as the default link type. A common use case for this is when linking to multiple resources.
```

Reviewer 2: > as the default link type.

Though, for clarity I think this should be link **color** as that's the correct property name (even if it is also a type).

Author: Changed it
